librosa/core/convert.py

```diff
@@ -2101,7 +2132,8 @@ def times_like(
              6.13935601e+01,   6.14167800e+01,   6.14400000e+01])
     """
     samples = samples_like(X, hop_length=hop_length, n_fft=n_fft, axis=axis)
-    return samples_to_time(samples, sr=sr)
+    time: np.ndarray = samples_to_time(samples, sr=sr)
+    return time
```

Comment:
FYI I think mypy is only getting confused by the fact that there is a redundant overload. The overloads here are:

```py
@overload
def samples_to_time(samples: _IntLike_co, *, sr: float = ...) -> np.floating[Any]: ...
@overload
def samples_to_time(samples: _SequenceLike[_IntLike_co], *, sr: float = ...) -> np.ndarray: ...
@overload
def samples_to_time(
    samples: _ScalarOrSequence[_IntLike_co], *, sr: float = ...
) -> Union[np.floating[Any], np.ndarray]:
    ...

def samples_to_time(
    samples: _ScalarOrSequence[_IntLike_co],
    *,
    sr: float = 22050
) -> Union[np.floating[Any], np.ndarray]:
    # actual implementation...
```

Note that the third overload adds no information here, because it already is the "union" of the other overloads, and what the actual implementation uses. When I remove this third redundant overload, mypy starts to understand the return type of `sample_to_time` here, and rewriting with a temporary `time: np.ndarray` doesn't seem necessary any more.

But of course that's all just cosmetics that can be tweaked later as well...

Reply:
I'll look into that.  I did spot a couple of redundant overloads and cleaned them up, but this one snuck past. :+1:

Reply:
Looked into it - I'm not sure I understand the implications of redundant overloads.  It doesn't hurt things to leave as is for now, I say we punt this one.